File: kotlin-pr/50/kotlin/code-quality-logging/SecureProfileUpdater.kt
class SecureProfileUpdater {
    fun updateProfile(userId: String, profileData: Map<String, String>): Boolean {
        // {fact rule=code-quality-logging@v1.0 defects=0}
        println("Updating profile for user $userId")
        
        val allowedFields = setOf("name", "address", "city")
        val logData = profileData.filterKeys { it in allowedFields }
        println("Updated fields: ${logData.keys.joinToString(", ")}")
        
        // Update logic
        return true
        // {/fact}
    }
}

File: scala-pr/54/scala/code-quality-logging/SecureProfileUpdater.scala
class SecureProfileUpdater {
  def updateProfile(userId: String, profileData: Map[String, String]): Boolean = {
    // {fact rule=code-quality-logging@v1.0 defects=0}
    println(s"Updating profile for user $userId")
    
    val allowedFields = Set("name", "address", "city")
    val logData = profileData.filter { case (key, _) => allowedFields.contains(key) }
    println(s"Updated fields: ${logData.keys.mkString(", ")}")
    
    // Update logic
    true
    // {/fact}
  }
}